src/services/search-service.ts

```diff
@@ -1,18 +1,45 @@
+import { RowDataPacket } from "mysql2";
 import logger from "../utils/logger/logger";
-
-// import SQLService from "./sql-service";
+import SQLService from "./sql-service";
 
 class SearchService {
-	static async search() {
-		logger.info("Search");
+	static async search(
+		table: string,
+		key: string,
+		columns: string[]
+	): Promise<RowDataPacket[]> {
+		logger.info(
+			`SearchService invoked! Table = ${table}, Key = ${key}, Columns = ${columns}`
+		);
+
+		const whereParams = ` WHERE MATCH (${columns})`;
+		const againstParams = ` AGAINST ('${key}' IN NATURAL LANGUAGE MODE)`;
```

Comment:
Does this SQL code create the index here in real-time? Or is the index created in the database before API usage?

Reply:
If the index is created in the DB beforehand, can you provide the SQL code to create the indices?

Reply:
I completely forgot, the SQL code does not create the index it only SELECTS from the index. I left index creation up to us so we can find tune when and how we create the index.

To create the index for **events** view the code snipped below.
```SQL
ALTER TABLE event ADD FULLTEXT(title, description);
```

To create the index for **contacts** view the code snipped below.
```SQL
ALTER TABLE contact ADD FULLTEXT(contact_id, uh_id, email, first_name, last_name, phone_number);
```
! Becareful as the ALTER code can be ran multiple times creating multiple indexes.

If for any reason there is a duplicate creation of an index or you would like to delete an index you can SHOW the table
```SQL
SHOW CREATE TABLE table_name;
```
and delete the `FULLTEXT KEY` index using this code 
```SQL
ALTER TABLE event DROP INDEX fulltext_key_name;
```